### PR TITLE
AVL tree in Java

### PR DESCRIPTION
**Pull request by sakshikhachane**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

AVL tree in Java. AVL tree is a self-balancing Binary Search Tree (BST).

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Solves : #2065

### CODE REVIEW

**Comment by sakshikhachane** [comment]
@atarax665 Please review the PR.

File: Java/ds/AVL_Tree.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
+/*
+    AVL tree is a self-balancing Binary Search Tree (BST) where the difference between heights of left and right subtrees cannot be more than one for all nodes and if at any time they differ by more than one, rebalancing is done to restore this property.
```
Comment (atarax665) [inline Java/ds/AVL_Tree.java]: Justify align this description part.

Comment (sakshikhachane) [reply Java/ds/AVL_Tree.java]: @atarax665 Is it okay now?

**Comment by sakshikhachane** [comment]
@atarax665 Are there any changes required?

File: Java/ds/AVL_Tree.java [hunk 2]
```diff
@@ -0,0 +1,198 @@
+/*
+AVL tree is a self-balancing Binary Search Tree (BST) where the difference between heights of left and right subtrees cannot be more than one for all nodes and if at any time they differ by more than one, rebalancing is done to restore this property. Each node maintains extra information called a balance factor whose value is either -1, 0 or +1 where balance factor of a node in an AVL tree is the difference between the height of the left subtree and that of the right subtree of that node.
```
Comment (HarshCasper) [inline Java/ds/AVL_Tree.java]: Break this line into multiple smaller lines and justify the content. Even on a Desktop, its difficult to read

**Review by HarshCasper — CHANGES_REQUESTED** [review]
A small change required. The rest looks good. 

Please ping me on Discord once the changes are done. Thanks!

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**